backend/app/rag/rag.py

```diff
@@ -71,6 +73,8 @@ def build_vector_db_retriever(populate_db: bool) -> VectorStoreRetriever:
         VectorStoreRetriever: A retriever instance for querying the vector database.
     """
     embedding = get_embedding_model(EMBEDDING_TYPE)
+    if not embedding:
+        return None
     vector_db = load_vector_db(embedding, TABLE_NAME)
```

Comment:
Here I'd also raise an error instead of returning None. Because without embedding the code will fail anyways at some point and we can have a good error message here

Reply:
Then you can also not return None again